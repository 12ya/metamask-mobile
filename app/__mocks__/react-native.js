import { Linking, DeviceEventEmitter } from 'react-native';

Linking.addEventListener = jest.fn();
Linking.removeEventListener = jest.fn((event, callback) => {
  // Simulate the behavior of removeEventListener
  if (typeof callback === 'function') {
    callback();
  }
});
Linking.openURL = jest.fn();
Linking.canOpenURL = jest.fn();
Linking.getInitialURL = jest.fn();

DeviceEventEmitter.addListener = jest.fn();
DeviceEventEmitter.removeListener = jest.fn((event, callback) => {
  // Simulate the behavior of removeListener
  if (typeof callback === 'function') {
    callback();
  }
});

export default { Linking, DeviceEventEmitter };
