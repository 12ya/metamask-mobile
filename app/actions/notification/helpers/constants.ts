enum notificationsAction {
  PERFORM_SIGN_IN_REQUEST = 'notifications/perform_sign_in_request',
  PERFORM_SIGN_IN_SUCCESS = 'notifications/perform_sign_in_success',
  PERFORM_SIGN_IN_FAILURE = 'notifications/perform_sign_in_failure',
  PERFORM_SIGN_OUT_REQUEST = 'notifications/perform_sign_out_request',
  PERFORM_SIGN_OUT_SUCCESS = 'notifications/perform_sign_out_success',
  PERFORM_SIGN_OUT_FAILURE = 'notifications/perform_sign_out_failure',
  ENABLE_PROFILE_SYNCING_REQUEST = 'notifications/enable_profile_syncing_request',
  ENABLE_PROFILE_SYNCING_SUCCESS = 'notifications/enable_profile_syncing_success',
  ENABLE_PROFILE_SYNCING_FAILURE = 'notifications/enable_profile_syncing_failure',
  DISABLE_PROFILE_SYNCING_REQUEST = 'notifications/disable_profile_syncing_request',
  DISABLE_PROFILE_SYNCING_SUCCESS = 'notifications/disable_profile_syncing_success',
  DISABLE_PROFILE_SYNCING_FAILURE = 'notifications/disable_profile_syncing_failure',
  ENABLE_PUSH_NOTIFICATIONS_REQUEST = 'notifications/enable_push_notifications_request',
  ENABLE_PUSH_NOTIFICATIONS_SUCCESS = 'notifications/enable_push_notifications_success',
  ENABLE_PUSH_NOTIFICATIONS_FAILURE = 'notifications/enable_push_notifications_failure',
  DISABLE_PUSH_NOTIFICATIONS_REQUEST = 'notifications/disable_push_notifications_request',
  DISABLE_PUSH_NOTIFICATIONS_SUCCESS = 'notifications/disable_push_notifications_success',
  DISABLE_PUSH_NOTIFICATIONS_FAILURE = 'notifications/disable_push_notifications_failure',
  CHECK_ACCOUNTS_PRESENCE_REQUEST = 'notifications/check_accounts_presence_request',
  CHECK_ACCOUNTS_PRESENCE_SUCCESS = 'notifications/check_accounts_presence_success',
  CHECK_ACCOUNTS_PRESENCE_FAILURE = 'notifications/check_accounts_presence_failure',
  DELETE_ON_CHAIN_TRIGGERS_BY_ACCOUNT_REQUEST = 'notifications/delete_on_chain_triggers_by_account_request',
  DELETE_ON_CHAIN_TRIGGERS_BY_ACCOUNT_SUCCESS = 'notifications/delete_on_chain_triggers_by_account_success',
  DELETE_ON_CHAIN_TRIGGERS_BY_ACCOUNT_FAILURE = 'notifications/delete_on_chain_triggers_by_account_failure',
  UPDATE_ON_CHAIN_TRIGGERS_BY_ACCOUNT_REQUEST = 'notifications/update_on_chain_triggers_by_account_request',
  UPDATE_ON_CHAIN_TRIGGERS_BY_ACCOUNT_SUCCESS = 'notifications/update_on_chain_triggers_by_account_success',
  UPDATE_ON_CHAIN_TRIGGERS_BY_ACCOUNT_FAILURE = 'notifications/update_on_chain_triggers_by_account_failure',
  SET_FEATURE_ANNOUNCEMENTS_ENABLED_REQUEST = 'notifications/set_feature_announcements_enabled_request',
  SET_FEATURE_ANNOUNCEMENTS_ENABLED_SUCCESS = 'notifications/set_feature_announcements_enabled_success',
  SET_FEATURE_ANNOUNCEMENTS_ENABLED_FAILURE = 'notifications/set_feature_announcements_enabled_failure',
  SET_SNAP_NOTIFICATIONS_ENABLED_REQUEST = 'notifications/set_snap_notifications_enabled_request',
  SET_SNAP_NOTIFICATIONS_ENABLED_SUCCESS = 'notifications/set_snap_notifications_enabled_success',
  SET_SNAP_NOTIFICATIONS_ENABLED_FAILURE = 'notifications/set_snap_notifications_enabled_failure',
  SET_METAMASK_NOTIFICATIONS_FEATURE_SEEN_REQUEST = 'notifications/set_metamask_notifications_feature_seen_request',
  SET_METAMASK_NOTIFICATIONS_FEATURE_SEEN_SUCCESS = 'notifications/set_metamask_notifications_feature_seen_success',
  SET_METAMASK_NOTIFICATIONS_FEATURE_SEEN_FAILURE = 'notifications/set_metamask_notifications_feature_seen_failure',
  FETCH_AND_UPDATE_METAMASK_NOTIFICATIONS_REQUEST = 'notifications/fetch_and_update_metamask_notifications_request',
  FETCH_AND_UPDATE_METAMASK_NOTIFICATIONS_SUCCESS = 'notifications/fetch_and_update_metamask_notifications_success',
  FETCH_AND_UPDATE_METAMASK_NOTIFICATIONS_FAILURE = 'notifications/fetch_and_update_metamask_notifications_failure',
  MARK_METAMASK_NOTIFICATIONS_AS_READ_REQUEST = 'notifications/mark_metamask_notifications_as_read_request',
  MARK_METAMASK_NOTIFICATIONS_AS_READ_SUCCESS = 'notifications/mark_metamask_notifications_as_read_success',
  MARK_METAMASK_NOTIFICATIONS_AS_READ_FAILURE = 'notifications/mark_metamask_notifications_as_read_failure',
  DELETE_NOTIFICATION_STATUS_REQUEST = 'notifications/delete_notification_status_request',
  DELETE_NOTIFICATION_STATUS_SUCCESS = 'notifications/delete_notification_status_success',
  DELETE_NOTIFICATION_STATUS_FAILURE = 'notifications/delete_notification_status_failure',
  SET_PARTICIPATE_IN_META_METRICS_REQUEST = 'notifications/set_participate_in_meta_metrics_request',
  SET_PARTICIPATE_IN_META_METRICS_SUCCESS = 'notifications/set_participate_in_meta_metrics_success',
  SET_PARTICIPATE_IN_META_METRICS_FAILURE = 'notifications/set_participate_in_meta_metrics_failure',
  UPDATE_TRIGGER_PUSH_NOTIFICATIONS_REQUEST = 'notifications/update_trigger_push_notifications_request',
  UPDATE_TRIGGER_PUSH_NOTIFICATIONS_SUCCESS = 'notifications/update_trigger_push_notifications_success',
  UPDATE_TRIGGER_PUSH_NOTIFICATIONS_FAILURE = 'notifications/update_trigger_push_notifications_failure',
  ENABLE_NOTIFICATIONS_SERVICES_REQUEST = 'notifications/enable_notifications_services_request',
  ENABLE_NOTIFICATIONS_SERVICES_SUCCESS = 'notifications/enable_notifications_services_success',
  ENABLE_NOTIFICATIONS_SERVICES_FAILURE = 'notifications/enable_notifications_services_failure',
  DISABLE_NOTIFICATIONS_SERVICES_REQUEST = 'notifications/disable_notifications_services_request',
  DISABLE_NOTIFICATIONS_SERVICES_SUCCESS = 'notifications/disable_notifications_services_success',
  DISABLE_NOTIFICATIONS_SERVICES_FAILURE = 'notifications/disable_notifications_services_failure',
}

export default notificationsAction;
