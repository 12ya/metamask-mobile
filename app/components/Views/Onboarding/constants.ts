// eslint-disable-next-line import/prefer-default-export
export const onboardingStrategy = {
  privateKey: 'privateKey',
  seedPhrase: 'seedPhrase',
  qrHardware: 'qrHardware',
};
