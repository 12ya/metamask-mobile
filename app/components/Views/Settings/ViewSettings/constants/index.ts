/* eslint-disable @typescript-eslint/no-var-requires */
/* eslint-disable @typescript-eslint/no-require-imports */
/* eslint-disable import/no-commonjs */
const VIEWS: Record<string, any> = {
  'MetaMask Default':
    require('../../../../../components/Views/Wallet/themes/01').default,
  'MetaMask NFTLover':
    require('../../../../../components/Views/Wallet/themes/02').default,
  'MetaMask web3Banking':
    require('../../../../../components/Views/Wallet/themes/03').default,
};

export default VIEWS;
