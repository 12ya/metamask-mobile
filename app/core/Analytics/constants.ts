export enum DeletionTaskStatus {
  pending = 'PENDING',
  staging = 'STAGING',
  started = 'STARTED',
  success = 'SUCCESS',
  failure = 'FAILURE',
  revoked = 'REVOKED',
  notFound = 'NOT_FOUND',
  unknown = 'UNKNOWN',
}

export enum ResponseStatus {
  ok = 'ok',
  error = 'error',
}

export enum States {
  enabled = 'ENABLED',
  disabled = 'DISABLED',
}

export const METAMETRICS_ANONYMOUS_ID = '0x0000000000000000';

export const SEGMENT_REGULATIONS_ENDPOINT =
  'https://platform.segmentapis.com/v1beta/workspaces/myworkspace/regulations';
