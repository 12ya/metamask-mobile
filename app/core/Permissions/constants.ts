export const CaveatTypes = Object.freeze({
  restrictReturnedAccounts: 'restrictReturnedAccounts',
});

export const RestrictedMethods = Object.freeze({
  eth_accounts: 'eth_accounts',
  ///: BEGIN:ONLY_INCLUDE_IF(snaps)
  // Snap Specific Restricted Methods
  snap_notify: 'snap_notify',
  snap_dialog: 'snap_dialog',
  snap_manageState: 'snap_manageState',
  snap_getBip32PublicKey: 'snap_getBip32PublicKey',
  snap_getBip32Entropy: 'snap_getBip32Entropy',
  snap_getBip44Entropy: 'snap_getBip44Entropy',
  snap_getEntropy: 'snap_getEntropy',
  wallet_snap: 'wallet_snap',
  ///: END:ONLY_INCLUDE_IF
});

export const NETWORK_TYPES = {
  GOERLI: 'goerli',
  LOCALHOST: 'localhost',
  MAINNET: 'mainnet',
  RPC: 'rpc',
  SEPOLIA: 'sepolia',
  LINEA_GOERLI: 'linea-goerli',
  LINEA_SEPOLIA: 'linea-sepolia',
  LINEA_MAINNET: 'linea-mainnet',
} as const;
