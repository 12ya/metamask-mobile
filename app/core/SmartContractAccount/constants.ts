export const POLYGON_MUMBAI_ADDRESS =
  '0x665cf455371e12EA5D49a7bA295cD060f436D95e';
export const ENTRY_POINT_ADDRESS = {
  mumbai: '0x5FF137D4b0FDCD49DcA30c7CF57E578a026d2789',
};
export const POLYGON_MUMBAI_CHAIN_ID = 80001;
export const POLYGON_MUMBAI_NAME = 'maticmum';
