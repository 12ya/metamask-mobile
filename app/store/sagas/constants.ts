/* eslint-disable import/prefer-default-export */
export const NOTIFICATIONS_ERRORS = {
  FAILED_TO_DELETE_NOTIFICATIONS: 'FAILED_TO_DELETE_NOTIFICATIONS',
  FAILED_TO_MARK_AS_READ_NOTIFICATIONS: 'FAILED_TO_MARK_AS_READ_NOTIFICATIONS',
  FAILED_TO_FETCH_NOTIFICATIONS: 'FAILED_TO_FETCH_NOTIFICATIONS',
  FAILED_TO_SET_NOTIFICATIONS_FEATURE_SEEN:
    'FAILED_TO_SET_NOTIFICATIONS_FEATURE_SEEN',
  FAILED_TO_CHECK_ACCOUNTS_PRESENCE: 'FAILED_TO_CHECK_ACCOUNTS_PRESENCE',
  FAILED_TO_ENABLE_PARTICIPATE_IN_META_METRICS:
    'FAILED_TO_ENABLE_PARTICIPATE_IN_META_METRICS',
  FAILED_TO_ENABLE_SNAP_NOTIFICATIONS: 'FAILED_TO_ENABLE_SNAP_NOTIFICATIONS',
  FAILED_TO_ENABLE_FEATURE_NOTIFICATIONS:
    'FAILED_TO_ENABLE_FEATURE_NOTIFICATIONS',
  FAILED_TO_DISABLE_METAMASK_NOTIFICATIONS:
    'FAILED_TO_DISABLE_METAMASK_NOTIFICATIONS',
  FAILED_TO_ENABLE_METAMASK_NOTIFICATIONS:
    'FAILED_TO_ENABLE_METAMASK_NOTIFICATIONS',
  FAILED_TO_DISABLE_PROFILE_SYNCING: 'FAILED_TO_DISABLE_PROFILE_SYNCING',
  FAILED_TO_ENABLE_PROFILE_SYNCING: 'FAILED_TO_ENABLE_PROFILE_SYNCING',
  FAILED_TO_SIGN_OUT: 'FAILED_TO_SIGN_OUT',
  FAILED_TO_SIGN_IN: 'FAILED_TO_SIGN_IN',
};
