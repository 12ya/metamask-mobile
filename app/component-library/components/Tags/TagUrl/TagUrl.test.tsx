// Third party dependencies.
import React from 'react';
import { render } from '@testing-library/react-native';

// Internal dependencies.
import TagUrl from './TagUrl';
import { TEST_IMAGE_SOURCE, TEST_LABEL } from './TagUrl.constants';

describe('TagUrl', () => {
  it('should render correctly', () => {
    const { toJSON } = render(
      <TagUrl imageSource={TEST_IMAGE_SOURCE} label={TEST_LABEL} />,
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
