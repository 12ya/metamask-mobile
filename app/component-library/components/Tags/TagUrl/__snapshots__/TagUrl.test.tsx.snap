// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagUrl should render correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "alignSelf": "center",
      "backgroundColor": "#FFFFFF",
      "borderColor": "#BBC0C5",
      "borderRadius": 99,
      "borderWidth": 1,
      "flexDirection": "row",
      "paddingLeft": 8,
      "paddingRight": 16,
      "paddingVertical": 8,
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "#FFFFFF",
        "borderRadius": 0,
        "height": 32,
        "marginRight": 8,
        "overflow": "hidden",
        "width": 32,
      }
    }
  >
    <Image
      onError={[Function]}
      resizeMode="contain"
      source={
        {
          "uri": "https://uniswap.org/favicon.ico",
        }
      }
      style={
        {
          "flex": 1,
          "height": undefined,
          "width": undefined,
        }
      }
      testID="favicon-avatar-image"
    />
  </View>
  <Text
    style={
      {
        "color": "#535A61",
        "flexShrink": 1,
        "fontFamily": "Euclid Circular B",
        "fontSize": 14,
        "fontWeight": "400",
        "letterSpacing": 0,
        "lineHeight": 22,
      }
    }
  >
    https://uniswap.org
  </Text>
</View>
`;
